### PR TITLE
Add ServerErrorHealthCheck

### LINKED ISSUE #43: Add ServerErrorHealthCheck

### PR DESCRIPTION
**Pull request by chrisrohr**
Closes #43

### CODE REVIEW

**Review by sleberknight — CHANGES_REQUESTED** [review]

File: src/main/java/org/kiwiproject/dropwizard/util/health/ServerErrorHealthCheck.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        if (isNullOrEmpty(meters) || !meters.containsKey(METER_NAME)) {
+            return newHealthyResult("%s meter is not configured", METER_NAME);
+        }
+
+        var meter = meters.get(METER_NAME);
+        var estimatedErrorCount = meter.getFifteenMinuteRate() * FIFTEEN_MINUTES_IN_SECONDS;
+
+        LOG.trace("15 minute rate on 5xx responses meter is roughly {}", estimatedErrorCount);
+
+        if (estimatedErrorCount >= criticalThreshold) {
```
[the remark this answers is not in the document]
Comment (chrisrohr) [reply src/main/java/org/kiwiproject/dropwizard/util/health/ServerErrorHealthCheck.java:74]: The original code set unhealthy if >= 1.0

File: src/test/java/org/kiwiproject/dropwizard/util/health/ServerErrorHealthCheckTest.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            assertThatHealthCheck(healthCheck)
+                    .isHealthy()
+                    .hasMessage("{} meter is not configured", METER_NAME);
+        }
+
+        @ParameterizedTest
+        @ValueSource(doubles = {0.0, 0.5, 0.999999})
+        void whenRequiredMeterIsReturnedAndRateIsLessThanThreshold(double rate) {
+            var meter = mock(Meter.class);
+            when(meter.getFifteenMinuteRate()).thenReturn(rate/FIFTEEN_MINUTES_IN_SECONDS);
```
[the remark this answers is not in the document]
Comment (chrisrohr) [reply src/test/java/org/kiwiproject/dropwizard/util/health/ServerErrorHealthCheckTest.java:71]: yes

File: src/test/java/org/kiwiproject/dropwizard/util/health/ServerErrorHealthCheckTest.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+            when(metrics.getMeters(METRIC_FILTER)).thenReturn(meters);
+
+            assertThatHealthCheck(healthCheck)
+                    .isHealthy()
+                    .hasMessage("{} meter is not configured", METER_NAME);
+        }
+
+        @ParameterizedTest
+        @ValueSource(doubles = {0.0, 0.5, 0.999999})
```
[the remark this answers is not in the document]
Comment (chrisrohr) [reply src/test/java/org/kiwiproject/dropwizard/util/health/ServerErrorHealthCheckTest.java:68]: Do you want me to add pioneer?

File: src/main/java/org/kiwiproject/dropwizard/util/health/ServerErrorHealthCheck.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (sleberknight) [reply src/main/java/org/kiwiproject/dropwizard/util/health/ServerErrorHealthCheck.java:74]: Understood, but it was originally done by someone who never bothered consulting anyone else, so was wondering if we want to change it.

File: src/test/java/org/kiwiproject/dropwizard/util/health/ServerErrorHealthCheckTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (sleberknight) [reply src/test/java/org/kiwiproject/dropwizard/util/health/ServerErrorHealthCheckTest.java:68]: Up to you...if you don't have time, then commit like this and I will take a break from all the retry library crap, I mean research, and add it myself. Actually let's do that 😜

File: src/main/java/org/kiwiproject/dropwizard/util/health/ServerErrorHealthCheck.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (chrisrohr) [reply src/main/java/org/kiwiproject/dropwizard/util/health/ServerErrorHealthCheck.java:74]: What do you want the defaults to be?  Remember too we can always change this to go against the actual rate not a number if wanted

[the remark this answers is not in the document]
Comment (chrisrohr) [reply src/main/java/org/kiwiproject/dropwizard/util/health/ServerErrorHealthCheck.java:88]: Made them ints... you can decide on new values

**Review by sleberknight — APPROVED** [review]

**Pull request merged**